elixir_rems_proxy/app.py

```diff
@@ -72,18 +66,19 @@
 async def user_patch(request):
     """PATCH request to the /user endpoint.
 
-    Update dataset permissions for given user.
+    Update user details or dataset permissions for given user.
     """
     LOG.debug('PATCH Request received.')
     db_pool = request.app['pool']
 
     if 'user' in request.match_info:
         user_identifier = request.match_info['user']
-        processed_request = await process_patch_request(user_identifier, request, db_pool)
-        if len(processed_request) == 0 or processed_request is True:
-            return web.HTTPOk(text='Successful operation')
+        missing_datasets = await process_patch_request(user_identifier, request, db_pool)
+        if not missing_datasets:
+            return web.HTTPNoContent()
         else:
-            return web.HTTPCreated(text=f'Following datasets are missing from REMS ({processed_request}), check "GET /user" endpoint for permissions')
+            return web.HTTPCreated(text=f'User was created, but following datasets are missing from REMS ({missing_datasets}), '
```

Comment:
I don't think this is the proper response, and this error should be raised in `create_dataset_permissions`, maybe a `404` because the datasets are missing, or `409`